File: app/src/main/java/com/vtrack/util/FormatUtil.kt
package com.vtrack.util

import java.text.NumberFormat
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

object FormatUtil {
    private val currencyFormat = NumberFormat.getCurrencyInstance(Locale.US)
    private val dateFormat = SimpleDateFormat("MMM d, yyyy", Locale.US)
    private val numberFormat = NumberFormat.getNumberInstance(Locale.US).apply {
        maximumFractionDigits = 1
    }

    fun formatCurrency(amount: Double): String = currencyFormat.format(amount)
    fun formatDate(epochMillis: Long): String = dateFormat.format(Date(epochMillis))
    fun formatMpg(mpg: Double): String = "${numberFormat.format(mpg)} MPG"
    fun formatMiles(miles: Int): String = "${NumberFormat.getIntegerInstance(Locale.US).format(miles)} mi"
    fun formatGallons(gallons: Double): String = "${numberFormat.format(gallons)} gal"
}

File: vtrack-ios/VTrack/Utilities/FormatUtil.swift
import Foundation

enum FormatUtil {
    private static let currencyFormatter: NumberFormatter = {
        let f = NumberFormatter()
        f.numberStyle = .currency
        f.locale = Locale(identifier: "en_US")
        return f
    }()

    private static let dateFormatter: DateFormatter = {
        let f = DateFormatter()
        f.dateFormat = "MMM d, yyyy"
        return f
    }()

    static func formatCurrency(_ value: Double) -> String {
        currencyFormatter.string(from: NSNumber(value: value)) ?? "$0.00"
    }

    static func formatDate(_ date: Date) -> String {
        dateFormatter.string(from: date)
    }

    static func formatMpg(_ value: Double) -> String {
        String(format: "%.1f MPG", value)
    }

    static func formatMiles(_ value: Int) -> String {
        let formatter = NumberFormatter()
        formatter.numberStyle = .decimal
        return (formatter.string(from: NSNumber(value: value)) ?? "\(value)") + " mi"
    }

    static func formatGallons(_ value: Double) -> String {
        String(format: "%.1f gal", value)
    }
}
